src/EventProducerService.cpp

```diff
@@ -0,0 +1,217 @@
+// Copyright (C) Martin Da Costa 2023 (martindc.merg@gmail.com)
+// This file is part of VLCB-Arduino project on https://github.com/SvenRosvall/VLCB-Arduino
+// Licensed under the Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License.
+// The full licence can be found at: http://creativecommons.org/licenses/by-nc-sa/4.0/
+
+// TODO: Add AREQ and ASRQ opcode support.
+// Check error messages
+// Set Params flags
+
+#include <Streaming.h>
+#include "EventProducerService.h"
+#include <Controller.h>
+#include <vlcbdefs.hpp>
+
+namespace VLCB {
+
+void EventProducerService::setController(Controller *cntrl) 
+{
+  this->controller = cntrl;
+  this->module_config = cntrl->module_config;
+}
+
+void EventProducerService::begin()
+{
+  //Initialise instantMode
+  instantMode = module_config->currentMode;
+  //DEBUG_SERIAL << F("> instant MODE initialise as: ") << instantMode << endl;
+  
+  if (instantMode == MODE_UNINITIALISED)
+  {
+    return;
+  }
+  else
+  {
+    setProducedEvents();
+  }
+}
+
+void EventProducerService::setProducedEvents()
```

Comment:
Is this setting up own consumed events?
In that case I would rename it setupOwnEvents().
Also you could check the parameter flags with PF_COE. If the module hasn't set this then there is no need to fill this table.

Reply:
No. Produced events are now in the events table so that they can be taught.  I haven't done consume own events yet.

Reply:
Ah.